packages/video_player/video_player_web/lib/video_player_web.dart

```diff
@@ -0,0 +1,196 @@
+import 'dart:async';
+import 'dart:html';
+import 'dart:ui' as ui;
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+import 'package:video_player_platform_interface/video_player_platform_interface.dart';
+
+/// The web implementation of [VideoPlayerPlatform].
+///
+/// This class implements the `package:video_player` functionality for the web.
+class VideoPlayerPlugin extends VideoPlayerPlatform {
+  /// Registers this class as the default instance of [VideoPlayerPlatform].
+  static void registerWith(Registrar registrar) {
+    VideoPlayerPlatform.instance = VideoPlayerPlugin();
+  }
+
+  Map<int, _VideoPlayer> _videoPlayers = <int, _VideoPlayer>{};
+
+  int _textureCounter = 1;
+
+  @override
+  Future<void> init() async {
+    return _disposeAllPlayers();
+  }
+
+  @override
+  Future<void> dispose(int textureId) async {
+    _videoPlayers[textureId].dispose();
+    _videoPlayers.remove(textureId);
+    return null;
+  }
+
+  void _disposeAllPlayers() {
+    _videoPlayers
+        .forEach((_, _VideoPlayer videoPlayer) => videoPlayer.dispose());
+    _videoPlayers.clear();
+  }
+
+  @override
+  Future<int> create(DataSource dataSource) async {
+    final int textureId = _textureCounter;
+    _textureCounter++;
+
+    final _VideoPlayer player = _VideoPlayer(
+      uri: Uri.parse(dataSource.uri),
+      textureId: textureId,
+    );
+
+    player.setupVideoPlayer();
+
+    _videoPlayers[textureId] = player;
+    return textureId;
+  }
+
+  @override
+  Future<void> setLooping(int textureId, bool looping) async {
+    return _videoPlayers[textureId].setLooping(looping);
+  }
+
+  @override
+  Future<void> play(int textureId) async {
+    return _videoPlayers[textureId].play();
+  }
+
+  @override
+  Future<void> pause(int textureId) async {
+    return _videoPlayers[textureId].pause();
+  }
+
+  @override
+  Future<void> setVolume(int textureId, double volume) async {
+    return _videoPlayers[textureId].setVolume(volume);
+  }
+
+  @override
+  Future<void> seekTo(int textureId, Duration position) async {
+    return _videoPlayers[textureId].seekTo(position);
+  }
+
+  @override
+  Future<Duration> getPosition(int textureId) async {
+    _videoPlayers[textureId].sendBufferingUpdate();
+    return _videoPlayers[textureId].getPosition();
+  }
+
+  @override
+  Stream<VideoEvent> videoEventsFor(int textureId) {
+    return _videoPlayers[textureId].eventController.stream;
+  }
+
+  @override
+  Widget buildView(int textureId) {
+    return HtmlElementView(viewType: textureId.toString());
```

Comment:
Let's make the viewType something like `videoPlayer-$textureId`

Reply:
Done